Fix testmasm, testair, testb3, etc. after 263139@main.

MenloDorian opened the pull request:
#### e98791bbdde841dd27f2df819eb2c0aa0994ef77
<pre>
Fix testmasm, testair, testb3, etc. after 263139@main.
<a href="https://bugs.webkit.org/show_bug.cgi?id=256145">https://bugs.webkit.org/show_bug.cgi?id=256145</a>
rdar://108707654

Reviewed by Elliott Williams and Yusuke Suzuki.

263139@main (for Apple internal builds) made it such that JSC will only enable the JIT on Mac
if the appropriate entitlement is available.  This unfortunately causes testmasm, testair,
testb3, testdfg, and others to all stop being able to enable the JIT.  This patch fixes it so
that all the relevant test executables gets the entitlement they need to enable the JIT.

* Source/JavaScriptCore/Configurations/TestExecutable.xcconfig: Renamed from Source/JavaScriptCore/Configurations/TestAPI.xcconfig.
* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/JavaScriptCore/Scripts/process-entitlements.sh:

Canonical link: <a href="https://commits.webkit.org/263568@main">https://commits.webkit.org/263568@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6471cf1c6b1d1027cc0f56e2b59fb5be08e7c02b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/5092 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/5223 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/5400 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/6613 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/5177 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/5422 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/5199 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/5425 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/5179 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/5263 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/4570 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/6629 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/2750 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/4568 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/9798 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/4214 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/4613 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/4641 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/6251 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/4685 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/5057 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/4152 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/5186 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/4540 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/1303 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/8620 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/5330 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/573 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/4905 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/1418 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on Source/JavaScriptCore/Scripts/process-entitlements.sh:
```diff
@@ -112,23 +112,47 @@ elif [[ "${WK_PLATFORM_NAME}" == macosx ]]
+          "${PRODUCT_NAME}" == testdfg ||
+          "${PRODUCT_NAME}" == testmasm ||
+          "${PRODUCT_NAME}" == testmem ||
+          "${PRODUCT_NAME}" == testRegExp ]]; then mac_process_jsc_entitlements
```
typo?
```suggestion
          "${PRODUCT_NAME}" == testRegExp ]]; then maccatalyst_process_jsc_entitlements
```

MenloDorian replied:
Good catch.  Fixed.

emw-apple commented on Source/JavaScriptCore/Configurations/TestExecutable.xcconfig:
This is the same as TestAPI.xcconfig, can you remove that file and update the testapi target to reference this config too?

MenloDorian replied:
Good catch.  Will do.

emw-apple commented on Source/JavaScriptCore/Configurations/TestExecutable.xcconfig:
```diff
@@ -0,0 +1,49 @@
+// Explicitly add the PrivateHeaders directory to the search path so that generated header files can be found in production builds.
+HEADER_SEARCH_PATHS = "${BUILT_PRODUCTS_DIR}/DerivedSources/JavaScriptCore" "$(JAVASCRIPTCORE_FRAMEWORKS_DIR)/JavaScriptCore.framework/PrivateHeaders" $(inherited);
+
+OTHER_LDFLAGS = $(inherited) $(SOURCE_VERSION_LDFLAGS);
```
Duplicated L44, please remove.

MenloDorian replied:
Removed.

emw-apple commented on Source/JavaScriptCore/Scripts/process-entitlements.sh:
```diff
@@ -112,23 +112,47 @@ elif [[ "${WK_PLATFORM_NAME}" == macosx ]]
     [[ "${RC_XBS}" != YES ]] && plistbuddy Add :com.apple.security.get-task-allow bool YES
 
-    if [[ "${PRODUCT_NAME}" == jsc ]]; then mac_process_jsc_entitlements
+    if [[ "${PRODUCT_NAME}" == jsc ||
+          "${PRODUCT_NAME}" == dynbench ||
+          "${PRODUCT_NAME}" == minidom ||
+          "${PRODUCT_NAME}" == testair ||
+          "${PRODUCT_NAME}" == testb3 ||
+          "${PRODUCT_NAME}" == testdfg ||
+          "${PRODUCT_NAME}" == testmasm ||
+          "${PRODUCT_NAME}" == testmem ||
+          "${PRODUCT_NAME}" == testRegExp ]]; then mac_process_jsc_entitlements
     elif [[ "${PRODUCT_NAME}" == testapi ]]; then mac_process_testapi_entitlements
```
I'm a little curious why the other test targets do not use the `testapi` entitlements path. AFAICT, the only difference is that testapi gets a `com.apple.rootless.storage.JavaScriptCore` entitlement and jsc does not.

MenloDorian replied:
Yes, `com.apple.rootless.storage.JavaScriptCore` is the only difference.  I was erring on the side of since these tests didn't need it, I chose to go with the `jsc` entitlements which leave it out.

Constellation reviewed (approved):
r=me

Commit:
Fix testmasm, testair, testb3, etc. after 263139@main.
https://bugs.webkit.org/show_bug.cgi?id=256145
rdar://108707654

Reviewed by Elliott Williams and Yusuke Suzuki.

263139@main (for Apple internal builds) made it such that JSC will only enable the JIT on Mac
if the appropriate entitlement is available.  This unfortunately causes testmasm, testair,
testb3, testdfg, and others to all stop being able to enable the JIT.  This patch fixes it so
that all the relevant test executables gets the entitlement they need to enable the JIT.

* Source/JavaScriptCore/Configurations/TestExecutable.xcconfig: Renamed from Source/JavaScriptCore/Configurations/TestAPI.xcconfig.
* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/JavaScriptCore/Scripts/process-entitlements.sh:

Canonical link: https://commits.webkit.org/263568@main
